segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ConcatOperation.java

```diff
@@ -205,10 +205,10 @@ private void checkState() {
     }
 
     private void checkPreconditions() {
-        Preconditions.checkArgument(null != targetHandle, "targetHandle");
-        Preconditions.checkArgument(!targetHandle.isReadOnly(), "targetHandle");
-        Preconditions.checkArgument(null != sourceSegment, "targetHandle");
-        Preconditions.checkArgument(offset >= 0, "offset");
+        Preconditions.checkArgument(null != targetHandle, "targetHandle must not be null");
```

Comment:
Same here. See above.

Reply:
And everywhere else below.

Reply:
Do not want to change method signature and tests for this change.